swift/container/backend.py

```diff
@@ -822,101 +824,116 @@ def merge_pivots(item1, item2):
                     item2['prefixed'] = True
             return item2
 
-        def _merge_items_by_type(curs, rec_type='object'):
-            obj = rec_type == 'object'
-            if obj:
-                rec_list = item_list
-            else:
-                rec_list = pivot_range_list
-
+        def get_records_object(curs, offset, rec_list, query_mod):
+            chunk = [record['name'] for record
+                     in rec_list[offset:offset + SQLITE_ARG_LIMIT]]
+            sql = (('SELECT name, storage_policy_index, created_at '
+                    'FROM object '
+                    'WHERE %s name IN (%s)')
+                   % (query_mod, ','.join('?' * len(chunk))))
+            return (((record[0], record[1]), record[2])
+                    for record in curs.execute(sql, chunk))
+
+        def get_records_pivot(curs, offset, rec_list, query_mod):
+                lower = []
+                upper = []
+                for record in rec_list[offset:offset + SQLITE_ARG_LIMIT]:
+                    lower.append(record['lower'])
+                    upper.append(record['upper'])
+                # None/NULL must be checked with the IS operator
+                sql = (('SELECT lower, upper, 0, created_at '
+                        'FROM pivot_ranges '
+                        'WHERE %s(lower IN (%s) OR lower IS NULL)'
+                        ' AND (upper IN (%s) OR upper IS NULL)')
+                       % (query_mod,
+                          ','.join('?' * len(lower)),
+                          ','.join('?' * len(upper))))
+                return (((rec[0], rec[1], rec[2]), rec[3])
+                        for rec in curs.execute(sql, lower + upper))
+
+        def _merge_items(curs, rec_list, is_object=True):
+            query_mod = ''
             if self.get_db_version(conn) >= 1:
                 query_mod = ' deleted IN (0, 1) AND '
+
+            if is_object:
+                get_records = get_records_object
+                keys = ('name', 'storage_policy_index')
+                del_keys = ('name', 'storage_policy_index')
+                add_keys = ('name', 'created_at', 'size', 'content_type',
+                            'etag', 'deleted', 'storage_policy_index')
+                def transform(item, other):
+                    return max(item, other, key=lambda i: i['created_at'])
+                table = 'object'
             else:
-                query_mod = ''
+                get_records = get_records_pivot
+                keys = ('lower', 'upper', 'storage_policy_index')
+                del_keys = ('lower', 'upper')
+                add_keys = ('lower', 'created_at', 'upper', 'object_count',
+                            'bytes_used', 'deleted')
+                def transform(item, other):
+                    return merge_pivots(other, item)
```

Comment:
The merge_pivots method is already in the form of transform, so rather then a few sub-sub (inception of defs) we can at least simplify and remove one right?

transform = merge_pivots
  OR 
to_add_trasnform = merge_items

But the latter would mean you'd need to  move and set to_add_transform on line 867 for object_lists.